### PR TITLE
[url_launcher] Publish fully federated version

### PR DESCRIPTION
**Pull request by stuartmorgan**
Restores the app-facing package to a publishable state now that _android
and _ios implementation packages are published.

Part of flutter/flutter#68498

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by mvanbeusekom — APPROVED** [review]
LGTM, CI is all red but it just seems a lot of timeouts occurring which is not related to this PR.

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
Having a closer look, packages `url_launcher_ios` version 2.0.16 and `url_launcher_android` version 2.0.16 don't exists. On pub.dev they are reported as version 6.0.13.

**Comment by stuartmorgan** [comment]
🤦🏻‍♂️ Sorry, I didn't realize I hadn't ever checked the test results on this PR.

File: packages/url_launcher/url_launcher/pubspec.yaml [hunk 1]
```diff
@@ -32,11 +29,8 @@ dependencies:
   flutter:
     sdk: flutter
   meta: ^1.3.0
-  # Temporary path dependencies to allow moving Android and iOS implementations.
-  url_launcher_android:
-    path: ../url_launcher_android
-  url_launcher_ios:
-    path: ../url_launcher_ios
+  url_launcher_android: ^2.0.13
+  url_launcher_ios: ^2.0.13
```
Comment (mvanbeusekom) [inline packages/url_launcher/url_launcher/pubspec.yaml]: One more little mistake ;) The major number also needs to change. The version on pub.dev is `6.0.13`:

```suggestion
  url_launcher_android: ^6.0.13
  url_launcher_ios: ^6.0.13
```

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]

**Review by mvanbeusekom — APPROVED** [review]
Three times is the charm, LGTM ;)

File: packages/url_launcher/url_launcher/pubspec.yaml (hunk 1 above)
Comment (stuartmorgan) [reply packages/url_launcher/url_launcher/pubspec.yaml]: Yep, I was focused on the patch version being wrong and totally missed that the major version was completely different from the other implementation packages.

**Pull request merged**